packages/eslint-plugin/src/rules/consistent-type-assertions.ts

```diff
@@ -126,9 +124,10 @@ export default createRule<Options, MessageIds>({
         fix:
           messageId === 'as'
             ? (fixer): TSESLint.RuleFix => {
-                const tsNode = parserServices.esTreeNodeToTSNodeMap.get(
-                  node as TSESTree.TSTypeAssertion,
-                );
+                const tsNode = getParserServices(
```

Comment:
```suggestion
                // lazily access parserServices to avoid crashing on non TS files (#9860)
                const tsNode = getParserServices(
```